tests/functional/multi-modules/index.js

```diff
@@ -16,39 +16,34 @@ pass.on('data', data => {
     logBuffer.records.push(JSON.parse(data.toString().trim()));
 });
 
-describe('Config is shared and unique within one program', () => {
-    it('should find all log entries in the RingBuffer with the right ' +
-       'module name', done => {
+describe('Configs are independent for each logger', () => {
```

Comment:
The point of my change was that config would be independant for each API instance (and their `children` loggers); not for each logger.
I defended that view with @LaurenSpiegel, but I may have been mis-understood; so let me attempt to re-explain my stance here:
 * We need to have some sharing of the configuration, if we want to easily change the logging levels for a whole process (ex: MetaData's repd, and that's actually an ongoing work done by @lamphamsy if I'm not mis-remembering things)
 * We had some dependencies updating the configuration through the Logger constructor (actually the issue raised by you and @LaurenSpiegel , that led to these changes)

Thus we need:
 * A layer where configs are still shared (hence the new `API` layer)
 * Changing the API so that the `Logger` constructor no longer updates the config, so that dependencies like bucketclient or utapi do not screw the file-configured logging (hence my changes on the Logger constructor, along with the new `API`)

Then my changes proposed the following:
 * An API layer from which all spawned Loggers share the same config (all loggers instanciated through the pre-bound `Logger` constructor provided by the `API` instance would share the same configuration instance)
 * A per-API configuration (each new API instance has a different configuration)